### PR TITLE
fixes old autocoder printf types to newly setup types

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Fixes the old autocoder output to use correct `printf` output specifiers for the new types system.

### CODE REVIEW

File: Autocoders/Python/src/fprime_ac/generators/templates/component/cpp.tmpl [hunk 1]
```diff
@@ -474,7 +474,7 @@ namespace ${namespace} {
       (void) snprintf(
           portName,
           sizeof(portName),
-          "%s_${instance}_OutputPort[%d]",
+          "%s_${instance}_OutputPort[%" PRI_FwIndexType "]",
```
Comment (bocchino) [inline Autocoders/Python/src/fprime_ac/generators/templates/component/cpp.tmpl:477]: Similar comment here and for the rest of the changes. Make sure that in each case, the type of the printf specifier matches the type of the value.

**Review by bocchino — CHANGES_REQUESTED** [review]
See comments. The type of the printf specifier needs to match the type of the value.

File: Autocoders/Python/src/fprime_ac/generators/templates/component/cpp.tmpl [hunk 2]
```diff
@@ -447,7 +447,7 @@ namespace ${namespace} {
       (void) snprintf(
           portName,
           sizeof(portName),
-          "%s_${instance}_InputPort[%d]",
+          "%s_${instance}_InputPort[%" PRI_FwIndexType "]",
```
[the remark this answers is not in the document]
Comment (bocchino) [reply Autocoders/Python/src/fprime_ac/generators/templates/component/cpp.tmpl:450]: Or you could use PRI_FwNativeIntType here.

**Review by gjwatney — APPROVED** [review]

**Review by bocchino — APPROVED** [review]
Looks good!

**Review by gjwatney — APPROVED** [review]

**Pull request merged**